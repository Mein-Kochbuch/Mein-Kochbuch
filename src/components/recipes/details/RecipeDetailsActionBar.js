import React from "react";
import styled from "styled-components/native";
import {Rating} from "react-native-ratings";
import {useColorScheme} from "react-native-appearance";
import Colors from "react-native/Libraries/NewAppScreen/components/Colors";

export default function RecipeDetailsActionBar() {
    const isDarkMode = useColorScheme() === 'dark';

    return (
        <ActionBar>
            <RatingStyled
                type={"custom"}
                tintColor={isDarkMode ? Colors.darker : Colors.lighter}
            />
        </ActionBar>
    )
}
const RatingStyled = styled(Rating)`
  padding: 12px;
  width: 250px;
`

const ActionBar = styled.View`
  display: flex;
  margin: 12px;
  height: 100px;
  border-radius: 10px;
  border-width: 1px;
  border-color: #fff;
`
