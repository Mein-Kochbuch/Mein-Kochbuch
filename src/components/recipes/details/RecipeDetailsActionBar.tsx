import React from 'react';
import styled from 'styled-components/native';
import RatingCompontent from './RatingCompontent';
import Icon from 'react-native-vector-icons/Feather';
import {Appearance} from 'react-native';
// @ts-ignore
import Colors from 'react-native/Libraries/NewAppScreen/components/Colors';

interface RecipeDetailsActionBarProps {
  avgRating: number;
  ratingCount: number;
  ownRating?: number;
  favorite: boolean;
  handleRating: (rating: number) => void;
  handleFavorize: () => void;
}

export default function RecipeDetailsActionBar({
  avgRating,
  ratingCount,
  ownRating,
  handleRating,
  favorite,
  handleFavorize,
}: RecipeDetailsActionBarProps) {
  const isDarkMode = Appearance.getColorScheme() === 'dark';
  const dislikeColor = isDarkMode ? Colors.lighter : Colors.darker;

  return (
    <ActionBar>
      <RatingCompontent
        ownRating={ownRating}
        avgRating={avgRating}
        ratingCount={ratingCount}
        handleRating={handleRating}
      />
      <Icon
        name={'heart'}
        size={35}
        color={favorite ? '#F10F0F' : dislikeColor}
        onPress={handleFavorize}
      />
    </ActionBar>
  );
}

const ActionBar = styled.View`
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  margin: 12px;
`;
