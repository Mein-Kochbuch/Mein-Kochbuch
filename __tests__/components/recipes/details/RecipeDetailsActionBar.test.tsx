import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import RecipeDetailsActionBar from '../../../../src/components/recipes/details/RecipeDetailsActionBar';
import {render} from '@testing-library/react-native';
import Icon from 'react-native-vector-icons/Feather';

import Colors from 'react-native/Libraries/NewAppScreen/components/Colors';
import RatingComponent from '../../../../src/components/recipes/details/RatingComponent';

jest.mock('../../../../src/components/recipes/details/RatingComponent', () => {
  return props => {
    return <div {...props} testID="RatingComponent" />;
  };
});

jest.mock('react-native-vector-icons/Feather', () => {
  return props => {
    return <div {...props} testID="Icon" />;
  };
});

describe('RatingComponent', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Favorite True', () => {
    const handleRating = props => {};
    const handleFavorize = () => {};

    const component = render(
      <RecipeDetailsActionBar
        averageRating={3}
        ratingCount={17}
        ownRating={5}
        favorite={true}
        handleRating={handleRating}
        handleFavorize={handleFavorize}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Favorite false light theme', () => {
    const handleRating = props => {};
    const handleFavorize = () => {};

    const component = render(
      <RecipeDetailsActionBar
        averageRating={3}
        ratingCount={17}
        ownRating={5}
        favorite={false}
        handleRating={handleRating}
        handleFavorize={handleFavorize}
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
