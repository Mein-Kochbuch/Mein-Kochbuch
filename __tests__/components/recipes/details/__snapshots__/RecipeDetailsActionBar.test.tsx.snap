// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingComponent Favorite True 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <div
    averageRating={3}
    handleRating={[Function]}
    ownRating={5}
    ratingCount={17}
    testID="RatingComponent"
  />
  <div
    color="#F10F0F"
    name="heart"
    onPress={[Function]}
    size={35}
    testID="Icon"
  />
</View>
`;

exports[`RatingComponent Favorite false light theme 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "marginBottom": 12,
        "marginLeft": 12,
        "marginRight": 12,
        "marginTop": 12,
      },
    ]
  }
>
  <div
    averageRating={3}
    handleRating={[Function]}
    ownRating={5}
    ratingCount={17}
    testID="RatingComponent"
  />
  <div
    color="#222"
    name="heart"
    onPress={[Function]}
    size={35}
    testID="Icon"
  />
</View>
`;
